cheroot/test/test_conn.py

```diff
@@ -1000,3 +1041,73 @@ def test_No_CRLF(test_client, invalid_terminator):
     expected_resp_body = b'HTTP requires CRLF terminators'
     assert actual_resp_body == expected_resp_body
     conn.close()
+
+
+def test_invalid_selected_connection(test_client, monkeypatch):  # noqa: D200,D205,D400,E501
+    """Test the error handling segment of
+    ``cheroot.connections.ConnectionManager.get_conn``.
+    """
+    class FaultySelect:
+        def __init__(self, original_select):
+            self.original_select = original_select
+            self.request_served = False
+            self.os_error_triggered = False
+
+        def __call__(self, timeout):
+            if self.request_served:
+                self.os_error_triggered = True
+                raise OSError('Error while selecting the client socket.')
+            else:
+                return self.original_select(timeout)
+
+    class FaultyGetMap:
+        def __init__(self, original_get_map):
+            self.original_get_map = original_get_map
+            self.sabotage_conn = False
+            self.socket_closed = False
+
+        def __call__(self):
+            if self.sabotage_conn:
+                for _, (*_, conn) in self.original_get_map().items():
+                    if isinstance(conn, cheroot.server.HTTPConnection):
+                        # close the socket to cause OSError
+                        conn.close()
+                        self.socket_closed = True
```

Comment:
Let's deal with nesting a bit
```suggestion
            sabotage_targets = (
                conn for _, (*_, conn) in self.original_get_map().items()
                if isinstance(conn, cheroot.server.HTTPConnection)
            ) if self.sabotage_conn else ()

            for conn in sabotage_targets:
                # close the socket to cause OSError
                conn.close()
                self.socket_closed = True
```

Reply:
I really don't like the alternative that you are suggesting. To me is just complicating things for the impression of linearity for a generator that is guaranteed that will yield one element. Is a functional pattern in a non-functional code-base.

But! If you _really_ want that, then ok.